server/service/testing_utils.go

```diff
@@ -413,6 +414,11 @@ func RunServerForTestsWithDS(t *testing.T, ds fleet.Datastore, opts ...*TestServ
 		}
 	}
 
+	if len(opts) > 0 && opts[0].WithDEPWebview {
+		frontendHandler := WithMDMEnrollmentMiddleware(svc, logger, ServeFrontend("", false, logger), "")
```

Comment:
So, I'm not sure how the MDM enrollment gets called by the whole DEP flow, but if you don't need the full frontend server (it looks like you only care about ` "/mdm/sso"`?), could you just wrap `rootMux` instead and provide a no-op / mocked handler for the actual "/mdm/sso" response?